src/wallet/wallet2.cpp

```diff
@@ -2032,74 +3153,127 @@ bool wallet2::store_keys(const std::string& keys_file_name, const std::string& p
   account_data = buffer.GetString();
 
   // Encrypt the entire JSON object.
-  crypto::chacha_key key;
-  crypto::generate_chacha_key(password.data(), password.size(), key);
-
+  crypto::generate_chacha_key(password.data(), password.size(), key, m_kdf_rounds);
   std::string cipher;
   cipher.resize(account_data.size());
+
   keys_file_data.iv = crypto::rand<crypto::chacha_iv>();
-  crypto::chacha8(account_data.data(), account_data.size(), key, keys_file_data.iv, &cipher[0]);
+  crypto::chacha20(account_data.data(), account_data.size(), key, keys_file_data.iv, &cipher[0]);
   keys_file_data.account_data = cipher;
 
-  std::string buf;
-  r = ::serialization::dump_binary(keys_file_data, buf);
-  r = r && epee::file_io_utils::save_string_to_file(keys_file_name, buf); //and never touch wallet_keys_file again, only read
-  CHECK_AND_ASSERT_MES(r, false, "failed to generate wallet keys file " << keys_file_name);
+  unlock_keys_file();
+  r = ::serialization::dump_binary(keys_file_data, output_buffer);
+  lock_keys_file();
+
+  //unlock_keys_file();
+  //std::string buf;
+  //r = ::serialization::dump_binary(keys_file_data, buf);
+  //r = r && epee::file_io_utils::save_string_to_file(keys_file_name, buf); //and never touch wallet_keys_file again, only read
+  //CHECK_AND_ASSERT_MES(r, false, "failed to generate wallet keys file " << keys_file_name);
+  //lock_keys_file();
```

Comment:
@mbg033 Ask Andrew, he may help

Reply:
It looks like this code was note removed after refactoring.
And now it can be thrown away with no harm.
These instructions now placed right after the point where this function called.

Reply:
original commit: https://github.com/graft-project/GraftNetwork/commit/0d7b9f5db827beca094313661e6a1f7b36cb9362
this part was moved to `wallet2::store_keys`, no idea why it appeared here commented